compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -168,6 +168,9 @@ pub(super) struct DecodeContext<'a, 'tcx> {
 
     // Used for decoding interpret::AllocIds in a cached & thread-safe manner.
     alloc_decoding_session: Option<AllocDecodingSession<'a>>,
+
+    // decoded symbols from metadata
+    symbol_table: Option<FxHashMap<u32, Symbol>>,
```

Comment:
OK, `DecodeContext` is bad place to store that table, as it forces reading from file a lot. Where is a good place to save that table?